magda-web-client/src/Components/Common/DataPreviewMapOpenInNationalMapButton.tsx

```diff
@@ -0,0 +1,149 @@
+import { ParsedDistribution } from "helpers/record";
+
+type PropsType = {
+    distribution: ParsedDistribution;
+    buttonText: string;
+    style: {
+        [key: string]: any;
+    };
+};
+
+const DEFAULT_TARGET_URL = "https://nationalmap.gov.au/";
+
+class DataPreviewMapOpenInNationalMapButton extends Component<PropsType> {
+    private browser: BrowserDetectInfo;
+    private winRef: Window | null;
+
+    constructor(props) {
+        super(props);
+        this.state = {};
+        this.onPopUpMessageReceived = this.onPopUpMessageReceived.bind(this);
+        this.winRef = null;
+        this.browser = browser();
+    }
+
+    componentDidMount() {
+        if (this.browser.name === "ie" && this.browser?.versionNumber! < 12)
+            return;
+        window.addEventListener("message", this.onPopUpMessageReceived);
+    }
+
+    componentWillUnmount() {
+        if (this.browser.name === "ie" && this.browser?.versionNumber! < 12)
+            return;
+        window.removeEventListener("message", this.onPopUpMessageReceived);
+    }
+
+    createCatalogItemFromDistribution(withoutBaseMap = false) {
+        const { distribution } = this.props;
+        let catConfig;
+
+        if (config?.supportExternalTerriaMapV7 === true) {
+            catConfig = {
+                initSources: [
+                    {
+                        catalog: [
+                            {
+                                name: distribution?.title,
+                                type: "magda-item",
+                                distributionId: distribution?.identifier,
+                                url: config.baseExternalUrl,
+                                isEnabled: true,
+                                zoomOnEnable: true
+                            }
+                        ]
+                    }
+                ]
+            };
+        } else {
+            const dga_id_prefix = "data.gov.au-postMessage-";
+            catConfig = {
+                initSources: [
+                    {
+                        catalog: [
+                            {
+                                name: distribution?.title,
+                                type: "magda",
+                                recordId: distribution?.identifier,
+                                url: config.baseExternalUrl,
+                                id:
+                                    dga_id_prefix +
+                                    this.props.distribution.identifier
+                            }
+                        ],
+                        workbench: [
+                            dga_id_prefix + this.props.distribution.identifier
+                        ]
+                    }
+                ]
+            };
+        }
+
+        if (!withoutBaseMap) {
+            //--- will not set baseMap if pass config by URL
+            catConfig.initSources[0].baseMapName = "Positron (Light)";
+        }
+        return catConfig;
```

Comment:
This looks good. Note that no version 8 map will work with IE11 any more. Maybe the map preview shouldn't be shown if the combination of IE11 and site using v8 preview map is detected.

Reply:
Yeah make it only render the button when support v7 turned on or not IE 11